sycl/doc/extensions/proposed/sycl_ext_oneapi_complex.asciidoc

```diff
@@ -0,0 +1,467 @@
+
+            /*Complex operators*/
+
+            // operator=
+            template<X>
+            complex& operator+=(const complex<X>&);
+
+            // operator+=
+            template<X>
+            complex& operator+=(const complex<X>&);
+
+            // operator-=
+            template<X>
+            complex& operator-=(const complex<X>&);
+
+            // operator*=
+            template<X>
+            complex& operator*=(const complex<X>&);
+
+            // operator/=
+            template<X>
+            complex& operator/=(const complex<X>&);
+
+            /*Scalar operators*/
+
+            // operator=
+            complex& operator=(const T&);
+
+            // operator+=
+            complex& operator+=(const T&);
+
+            // operator-=
+            complex& operator-=(const T&);
+
+            // operator*=
+            complex& operator*=(const T&);
+
+            // operator/=
+            complex& operator/=(const T &);
+    }
+
+    /*Complex operators*/
+
+    // operator+
+    template<typename T>
+    complex<T> operator+(const complex<T>&, const complex<T>&);
+
+    // operator-
+    template<typename T>
+    complex<T> operator-(const complex<T>&, const complex<T>&);
+
+    // operator*
+    template<typename T>
+    complex<T> operator*(const complex<T>&, const complex<T>&);
+
+    // operator/
+    template<typename T>
+    complex<T> operator/(const complex<T>&, const complex<T>&);
+
+    // operator==
+    template<typename T>
+    bool operator==(const complex<T>&, const complex<T>&);
+
+    // operator!=
+    template<typename T>
+    bool operator!=(const complex<T>&, const complex<T>&);
+
+    /*Scalar operators*/
+
+    // operator+
+    template<typename T>
+    complex<T> operator+(const complex<T>&, const T&);
+    template<typename T>
+    complex<T> operator+(const T&, const complex<T>&);
+
+    // operator-
+    template<typename T>
+    complex<T> operator/(const complex<T>&, const T&);
+    template<typename T>
+    complex<T> operator/(const T>&, const complex<T>&);
+
+    // operator==
+    template<typename T>
+    bool operator==(const complex<T>&, const T&);
+    template<typename T>
+    bool operator==(const T&, const complex<T>&);
+
+    // operator!=
+    template<typename T>
+    bool operator!=(const complex<T>&, const T&);
+    template<typename T>
+    bool operator!=(const T&, const complex<T>&);
+
+    /*Stream operator*/
+
+    // operator<<
+    template<typename T>
+    const sycl::stream& operator<<(sycl::stream&, const complex<T>&);
+
+    // Note: The below << and >> operator with basic_ostream and basic_istream
+    // are host-only.
+    template<class T, class charT, class traits>
+    std::basic_ostream<charT, traits>&
+    operator<<(std::basic_ostream<charT, traits>&, const complex<T>&);
+
+    template<class T, class charT, class traits>
+    std::basic_istream<charT, traits>&
+    operator>>(std::basic_istream<charT, traits>&, const complex<T>&);
```

Comment:
To my knowledge we don't define these for other SYCL types, so I don't think we should for this either. It also creates a hard dependency on `istream` and `ostream` which we are currently trying to reduce.

Reply:
Thank you for the clarification. I have removed the `istream` and `ostream` operators, so there is no hard dependency in the extension proposal.

Reply:
They seem to still be here. Did you forget to push them or am I missing something?

Reply:
Sorry I forgot to push the change.